docs/typing.rst

```diff
@@ -34,3 +35,27 @@ A reference implementation of the SMIRNOFF specification for parameterizing biom
     :toctree: api/generated/
 
     ForceField
+
```

Comment:
Suggest we add more explanatory text here for the different groups of methods, and that we highlight the most important classes users will interact with (`ForceField`, and possibly `Parameter*` classes) before diving into the other detailed sections (`*Handler`) that only developers will be interacting with.

Reply:
Good idea. I've added two sentences of explanation to each section, added some formatting, and generally tried to improve the presentation of this page in 915b61d